core/dataset.cpp

```diff
@@ -124,7 +124,7 @@ DataConstProxy Dataset::operator[](const std::string_view name) const {
   if (it == m_data.end())
     throw std::out_of_range("Could not find data with name " +
                             std::string(name) + ".");
-  return DataConstProxy(*this, it->second);
+  return DataConstProxy(*this, it->second, std::string(name));
```

Comment:
```suggestion
  return DataConstProxy(*this, it);
```
Maybe simpler? `it` should contain all we need.